core/templates/domain/objects/UnitsObjectFactory.ts

```diff
@@ -264,6 +264,37 @@ export class UnitsObjectFactory {
     }
     return new Units(this.fromStringToList(units));
   }
+
+  getAllInstancesOfUnits(units: string[]): string[] {
+    // Returns a list with all the instances of unit in a units list.
+    var instancesOfUnits: string[] = [];
+    units.forEach(entry => {
+      if (entry.includes('^')) {
+        this.getAllInstancesOfUnits(entry.split('^')).forEach(entryWoutExp => {
+          instancesOfUnits.push(entryWoutExp);
+        });
+      } else if (this.isunit(entry) && entry.length > 0) {
+        instancesOfUnits.push(entry.replace(/[0-9]/g, ''));
+      }
+    });
+    return instancesOfUnits;
+  }
+
+  duplicatedUnit(units: string): string {
```

Comment:
Can we rename this to 
```suggestion
  getDuplicatedUnit(units: string): string {
```